pkg/kubelet/cm/dra/plugin/registration_test.go

```diff
@@ -130,16 +123,23 @@ func TestRegistrationHandler(t *testing.T) {
 				fakeClient.AddReactor("delete-collection", "resourceslices", func(action cgotesting.Action) (bool, runtime.Object, error) {
 					deleteAction := action.(cgotesting.DeleteCollectionAction)
 					restrictions := deleteAction.GetListRestrictions()
-					fieldsSelector := fields.SelectorFromSet(expectedSliceFields)
+					if !restrictions.Labels.Empty() {
+						return false, nil, fmt.Errorf("unexpected label selector: %s, expected empty selector", restrictions.Labels.String())
+					}
+
+					expectedFieldsSelector := fields.SelectorFromSet(expectedSliceFields)
 					// The order of field requirements is random because it comes
 					// from a map. We need to sort.
 					normalize := func(selector string) string {
 						requirements := strings.Split(selector, ",")
 						sort.Strings(requirements)
 						return strings.Join(requirements, ",")
 					}
-					assert.Equal(t, "", restrictions.Labels.String(), "label selector in DeleteCollection")
-					assert.Equal(t, normalize(fieldsSelector.String()), normalize(restrictions.Fields.String()), "field selector in DeleteCollection")
+					normalizedFieldsSelector := normalize(restrictions.Fields.String())
+					normalizedExpectedFieldsSelector := normalize(expectedFieldsSelector.String())
+					if normalizedFieldsSelector != normalizedExpectedFieldsSelector {
+						return false, nil, fmt.Errorf("unexpected field selector %s, expected: %s", normalizedFieldsSelector, normalizedExpectedFieldsSelector)
+					}
```

Comment:
How is returning an error here leading to a test failure?

Reply:
It's not. It does fix the test by ignoring (returning `false` from the reactor when it's called for `pluginA`.